### PR TITLE
storage/mysql: Fix detection of mysql enum column value changes

### PR DESCRIPTION
**Pull request by rjobanp**
### Motivation

  * This PR fixes a recognized bug. Fixes https://github.com/MaterializeInc/materialize/issues/26022

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [x] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [x] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [x] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by def- — APPROVED** [review]
testing side lgtm

File: src/mysql-util/src/desc.rs [hunk 1]
```diff
@@ -163,6 +167,33 @@ pub enum MySqlColumnMeta {
...
+                            && self_enum
+                                .values
+                                .iter()
+                                .zip(other_enum.values.iter())
+                                .all(|(self_val, other_val)| self_val == other_val)
+                    }
+                    (MySqlColumnMeta::Json, MySqlColumnMeta::Json) => true,
+                    (MySqlColumnMeta::Year, MySqlColumnMeta::Year) => true,
+                    _ => false,
+                }
```
Comment (guswynn) [inline src/mysql-util/src/desc.rs:191]:
```suggestion
            (None, None) => true,
            (Some(_), None) => false,
            (None, Some(_)) => false,
            (Some(MySqlColumnMeta::Enum(self_enum)), Some(MySqlColumnMeta::Enum(other_enum))) => {
                      // so as long as `self.values` is a compatible prefix of `other.values`, we can
                      // ignore extra values from `other.values`.
                      self_enum.values.len() <= other_enum.values.len()
                          && self_enum
                              .values
                              .iter()
                              .zip(other_enum.values.iter())
                              .all(|(self_val, other_val)| self_val == other_val)
                    }
               (Some(MySqlColumnMeta::Json), Some(MySqlColumnMeta::Json)) => true,     
               (Some(MySqlColumnMeta::Year), Some(MySqlColumnMeta::Year)) => true,     
                    _ => false,
                }
```

less nesting is probably clearer!

**Review by guswynn — APPROVED** [review]
Looks like you accidentally picked up some `Cargo.lock` changes that are causing the `cargo vet` test to fail

File: src/mysql-util/src/desc.rs (hunk 1 above)
Comment (rjobanp) [reply src/mysql-util/src/desc.rs:191]: ah yes great idea!

**Pull request merged**